nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -981,6 +1007,85 @@ where
     Ok(())
 }
 
+/// Collect task dumps from all SPs via MGS and save them to a directory.
+async fn save_all_sp_dumps(
+    log: &slog::Logger,
+    resolver: &Resolver,
+    sp_dumps_dir: &Utf8Path,
+) -> anyhow::Result<()> {
+    let mgs_client = resolver
+        .lookup_socket_v6(ServiceName::ManagementGatewayService)
+        .await
+        .map(|sockaddr| {
+            let url = format!("http://{}", sockaddr);
+            gateway_client::Client::new(&url, log.clone())
+        })
+        .context("failed to resolve address of MGS")?;
+
+    let all_sps = mgs_client
+        .sp_all_ids()
+        .await
+        .context("failed to get list of SPs from MGS")?
+        .into_inner();
+
+    let mut futures = futures::stream::iter(all_sps.into_iter())
```

Comment:
I recommend checking out `parallel-task-set`, which we added relatively recently, to perform this task-saving operation in parallel.

(It should be quite similar to using `buffer_unordered`, but it'll actually spawn tokio tasks for each SP - but we can still set a limit on the maximum amount of parallelism)